src/librustdoc/html/render/mod.rs

```diff
@@ -2410,3 +2429,206 @@ fn collect_paths_for_type(first_ty: clean::Type, cache: &Cache) -> Vec<String> {
+        let root_path = vec!["../"; cx.current.len() - 1].join("");
+
+        let mut decoration_info = FxHashMap::default();
+        decoration_info.insert("highlight", byte_ranges);
+
+        sources::print_src(
+            w,
+            contents_subset,
+            edition,
+            file_span,
+            cx,
+            &root_path,
+            Some(line_offset),
+            Some(highlight::DecorationInfo(decoration_info)),
+        );
+        write!(w, "</div></div>");
+
+        true
+    };
+
+    // The call locations are output in sequence, so that sequence needs to be determined.
+    // Ideally the most "relevant" examples would be shown first, but there's no general algorithm
+    // for determining relevance. Instead, we prefer the smallest examples being likely the easiest to
+    // understand at a glance.
+    let ordered_locations = {
+        let sort_criterion = |(_, call_data): &(_, &CallData)| {
+            // Use the first location because that's what the user will see initially
+            let (lo, hi) = call_data.locations[0].enclosing_item.byte_span;
+            hi - lo
+        };
+
+        let mut locs = call_locations.into_iter().collect::<Vec<_>>();
+        locs.sort_by_key(|x| sort_criterion(x));
+        locs
+    };
+
+    // Write just one example that's visible by default in the method's description.
+    let mut it = ordered_locations.into_iter().peekable();
+    let write_and_skip_failure = |w: &mut Buffer, it: &mut Peekable<_>| {
+        while let Some(example) = it.next() {
+            if write_example(&mut *w, example) {
+                break;
+            }
+        }
+    };
+
+    write_and_skip_failure(w, &mut it);
+
+    // Then add the remaining examples in a hidden section.
+    if it.peek().is_some() {
+        write!(
+            w,
+            r#"<details class="rustdoc-toggle more-examples-toggle">
+                  <summary class="hideme">
+                     <span>More examples</span>
+                  </summary>
+                  <div class="more-scraped-examples">
+                    <div class="toggle-line"><div class="toggle-line-inner"></div></div>
+                    <div class="more-scraped-examples-inner">
+"#
+        );
+
+        // Only generate inline code for MAX_FULL_EXAMPLES number of examples. Otherwise we could
+        // make the page arbitrarily huge!
+        for _ in 0..MAX_FULL_EXAMPLES {
+            write_and_skip_failure(w, &mut it);
+        }
+
+        // For the remaining examples, generate a <ul /> containing links to the source files.
+        if it.peek().is_some() {
+            write!(
+                w,
+                r#"<div class="example-links">Additional examples can be found in:<br /><ul>"#
+            );
+            it.for_each(|(_, call_data)| {
+                write!(
+                    w,
+                    r#"<li><a href="{}{}" target="_blank">{}</a></li>"#,
```

Comment:
If you're passing named formatters anyway, you may was well use them; or just remove the names, I think it's clear by context.
```suggestion
                    r#"<li><a href="{root}{url}" target="_blank">{name}</a></li>"#,
```